src/text-input/src/TextInput.js

```diff
@@ -88,25 +116,62 @@ class TextInput extends PureComponent {
     const borderRadius = theme.getBorderRadiusForControlHeight(height)
 
     return (
-      <Text
-        is="input"
-        className={cx(themedClassName, className)}
-        type="text"
-        size={textSize}
-        width={width}
+      <Box
```

Comment:
@jeroenransijn Here's what I am referring to as changing the shape, which I think causes `<div>` that breaks with the `FilePicker` snapshot test

Reply:
Yeah, this change is definitely tricky to make because we now have to wrap this component inside of component, while trying to mimic the API as if it was one single component.

I suggest taking a look at `SearchInput` as well, since we are doing almost the same there.